src/integrations/expressApollo.ts

```diff
@@ -0,0 +1,149 @@
+import * as express from 'express';
+import * as graphql from 'graphql';
+import { runQuery } from '../core/runQuery';
+
+import * as GraphiQL from '../modules/renderGraphiQL';
+
+// TODO: will these be the same or different for other integrations?
+export interface ExpressApolloOptions {
+  schema: graphql.GraphQLSchema;
+  formatError?: Function;
+  rootValue?: any;
+  context?: any;
+  logFunction?: Function;
+  formatRequest?: Function;
+  validationRules?: Array<Function>; // validation rules are functions
+  formatResponse?: Function;
+}
+
+export interface ExpressApolloOptionsFunction {
+  (req?: express.Request): ExpressApolloOptions | Promise<ExpressApolloOptions>;
+}
+
+// Design principles:
+// - there is just one way allowed: POST request with JSON body. Nothing else.
+// - simple, fast and secure
+//
+
+export interface ExpressHandler {
+  (req: express.Request, res: express.Response, next): void;
+}
+
+export function graphqlHTTP(options: ExpressApolloOptions | ExpressApolloOptionsFunction): ExpressHandler {
+  if (!options) {
+    throw new Error('Apollo Server requires options.');
+  }
+
+  if (arguments.length > 1) {
+    // TODO: test this
+    throw new Error(`Apollo Server expects exactly one argument, got ${arguments.length + 1}`);
+  }
+
+  return async (req: express.Request, res: express.Response, next) => {
```

Comment:
As far as I know, they're interchangeable, as suggested [here](https://ponyfoo.com/articles/understanding-javascript-async-await). Right now that gets transpiled to a promise anyway, if you look at the /dist folder.

The reason I used `async/await` is that it makes the code so much more readable. Compare our code to express-graphql, and you'll see what I mean.